idea/frontend/src/hooks/useLocalProgramsFilter.ts

```diff
@@ -0,0 +1,32 @@
+import { useApi } from '@gear-js/react-hooks';
+import { useEffect } from 'react';
+
+import { PROGRAMS_LOCAL_FORAGE } from 'api';
+import { IProgram } from 'entities/program';
+
+import { useChain } from './context';
+
+function useLocalProgramsFilter() {
+  const { api, isApiReady } = useApi();
+  const { isDevChain } = useChain();
+
+  const filterLocalPrograms = () => {
+    const genesis = api.genesisHash.toHex();
+
+    api.program.allUploadedPrograms().then((chainProgramIDs) =>
+      PROGRAMS_LOCAL_FORAGE.iterate((program: IProgram) => {
+        const isProgramInChain = chainProgramIDs.includes(program.id);
+        const isProgramFromChain = genesis === program.genesis;
+
+        if (!isProgramInChain || !isProgramFromChain) PROGRAMS_LOCAL_FORAGE.removeItem(program.id);
```

Comment:
```suggestion
        if (!isProgramInChain && !isProgramFromChain) PROGRAMS_LOCAL_FORAGE.removeItem(program.id);
```
I think you only need to remove the programs related to connected chain. What do you think?

Reply:
In this case if two programs have the same id, but different genesis, it won't be deleted.
There's no additional genesis check on programs render.